Azure Security Center (#81)
microsoft/azure_arc: azure_arc_servers_jumpstart/docs/arc_securitycenter.md

> # Azure Arc Machines on Azure Security Center

Reviewer: ```suggestion
# Azure Arc enabled Servers on Azure Security Center
```

# Azure Arc Machines on Azure Security Center

> The following README will guide you on how to onboard an Azure Arc enabled Machine on to [Azure Security Center (ASC)](https://docs.microsoft.com/en-us/azure/security-center/), it will collect security-related configurations and event logs to recommend actions to improve your security posture.
> 
> In this guide, you will enable Security Center on a subscription and implement the Standard tier, which provides advanced threat protection and detection capabilities. To complete this process you will: 
> 
> * Set the Log Analytics workspace where logs and events will be aggregated for analysis. 
> 
> * Assign Security Center’s default security policies.
> 
> * Review Azure Security Center's recommendations.
> 
> * Apply recommended configurations on Azure Arc enabled Machines using ***Quick Fix*** remediations. 

Reviewer: ```suggestion
The following README will guide you on how to onboard an Azure Arc enabled Machine on to [Azure Security Center (ASC)](https://docs.microsoft.com/en-us/azure/security-center/), so you can start collecting security-related configurations as well as event logs as a way to recommend actions and improve your overall Azure security posture.

In this guide, you will enable and configure Standard tier ASC on your Azure subscription, which will provide you with advanced threat protection (ATP) and detection capabilities. To complete this process you will: 

* Setup the Log Analytics Workspace where logs and events will be aggregated for analysis. 

* Assign Security Center’s default security policies.

* Review Azure Security Center's recommendations.

* Apply recommended configurations on Azure Arc enabled Servers using the ***Quick Fix*** remediations. 
```

**Note: This guide assumes you already deployed VMs or servers that are running on-premises or other clouds and you have connected them to Azure Arc.**

**If you haven't, this repository offers you a way to do so in an automated fashion:**
- **[GCP Ubuntu VM](gcp_terraform_ubuntu.md) / [GCP Windows VM](gcp_terraform_windows.md)**
- **[AWS Ubuntu VM](aws_terraform_ubuntu.md)**
- **[VMware Ubuntu VM](vmware_terraform_ubuntu.md) / [VMware Windows Server VM](vmware_terraform_winsrv.md)**
- **[Local Ubuntu VM](local_vagrant_ubuntu.md) / [Local Windows VM](local_vagrant_windows.md)**

# Prerequisites

* Clone this repo.

> * As mentioned, this guide starts at the point where you already deployed and connected VMs or servers to Azure Arc. In the screenshots below we can see a GCP server has been connected with Azure Arc and is visible as a resource in Azure.
> 

Reviewer: ```suggestion
* As mentioned, this guide starts at the point where you already deployed and connected VMs or bare-metal servers to Azure Arc. For this scenario, as can be seen in the screenshots below, we will be using a Google Cloud Platform (GCP) instance that has been already connected to Azure Arc and is visible as a resource in Azure.

```

  ```

* Select the Azure Security Center tier, the Free tier is enabled on all your Azure subscriptions by default and will provide continuous security assessment and actionable security recommendations. In this guide, you will use the Standard tier for Virtual Machines that extends these capabilities providing unified security management and threat protection across your hybrid cloud workloads. To enable the Standard tier of Azure Security Center for VMs run the command below: 

    ```bash
    az security pricing create -n VirtualMachines --tier 'standard'
    ```
* Now you need to assign the default Security Center policy initiative. ASC makes its security recommendations based on policies, there is an specific initiative that groups Security Center policies with the definition ID '1f3afdf9-d0c9-4c3d-847f-89da613e70a8'. The command below will assign the ASC initiative to your subscription:

    ```bash
    az policy assignment create --name 'ASC Default <Your subscription ID>' \
    --scope '/subscriptions/<Your subscription ID>' \
    --policy-set-definition '1f3afdf9-d0c9-4c3d-847f-89da613e70a8' 

>     ```

Reviewer: IMO, I think the Log Analytics workspace should be it's own section and not a prereq.

# Azure Security Center Integration

> Now that you have successfully onboarded Azure Security Center, you will get recommendations to help you protect your  resources, including your Azure Arc enabled Machines. Security Center will periodically analyze the security state of your Azure resources to identify potential security vulnerabilities.

Reviewer: ```suggestion
Now that you have successfully onboarded Azure Security Center, you will get recommendations to help you protect your resources, including your Azure Arc enabled Servers. Security Center will then periodically analyze the security state of your Azure resources to identify potential security vulnerabilities.
```

> * On the "Compute & Apps" section under "VM and Servers", Security Center will give you an overview of all security recommendations for your VMs and computers, including Azure VMs, Azure Classic VMs, servers and Azure Arc Machines. 

Reviewer: ```suggestion
* In the "Compute & apps" section under "VM and Servers", ASC will provide you with an overview of all the discovered security recommendations for your VMs and computers, including Azure VMs, Azure Classic VMs, servers and **Azure Arc Machines**. 
```

    ![](../img/securitycenter/04.png)

* On the Azure Arc machines, you will get a recommendation to install the log analytics agent.

>     ![](../img/securitycenter/05.png)

Reviewer: add red arrow next to the "Quick Fix"

> * On the Azure Arc machines, you will get a recommendation to install the log analytics agent.
> 
>     ![](../img/securitycenter/05.png)
> 
> * For each recommendation you will get: 
>     - A short description of what is being recommended.
>     - A Secure Score impact, in this case is High.
>     - The remediation steps to carry out in order to implement the recommendation. For specific recommendations, like this one, you get also a ***Quick Fix*** that enables you to quickly remediate a recommendation on multiple resources. 
>     
>     ![](../img/securitycenter/06.png)

Reviewer: ```suggestion
* On the Azure Arc machines, ASC will provide a recommendation to install the Log Analytics agent. In addition, each recommendation will include:
    - A short description of what is being recommended.
    - A Secure Score impact, in this case, set to *High*.
    - The remediation steps to carry out in order to implement the recommendation. For specific recommendations, like this one, you will also get a ***Quick Fix*** that enables you to quickly remediate a recommendation on multiple resources. 

    ![](../img/securitycenter/05.png)
    
    ![](../img/securitycenter/06.png)
```

>     ![](../img/securitycenter/06.png)

Reviewer: add red arrow next to the "Quick Fix"
